eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/IntInterval.java

```diff
@@ -1007,4 +1016,82 @@ public int next()
             throw new NoSuchElementException();
         }
     }
+
+    private static final class IntIntervalSpliterator implements Spliterator.OfInt
+    {
+        private int current;
+        private final int to;
+        private final int step;
+        private final boolean isAscending;
+
+        private IntIntervalSpliterator(int from, int to, int step)
+        {
+            this.to = to;
+            this.step = step;
+            this.current = from;
+            this.isAscending = from <= to;
+        }
+
+        @Override
+        public Comparator<? super Integer> getComparator()
+        {
+            return  new Comparator<Integer>()
+            {
+                @Override
+                public int compare(Integer o1, Integer o2)
+                {
+                    if (IntIntervalSpliterator.this.isAscending)
+                    {
+                        return o1 - o2;
+                    }
+                    return o2 - o1;
+                }
+            };
+        }
+
+        @Override
+        public OfInt trySplit()
+        {
+            if (this.isAscending)
+            {
+                int mid = this.current + (this.to - this.current) / (2 * this.step);
+                return (this.current >= mid)
+                        ? null
+                        : new IntIntervalSpliterator(this.current, mid, this.step);
+            }
+
+            int mid = this.current + (this.current - 9) / (2 * this.step);
```

Comment:
Where does the number 9 come from? Did you check code coverage? This is the most complex part of the whole patch, so ideally every line of this method has test coverage.